File: 121. Best time to buy and sell stock/soln.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int lp=0, rp=1, res = 0;
        int n = prices.size();
        if (n<2){
            return 0;
        }
        while(rp<n){
            res = max(res, prices[rp]-prices[lp]);
            if(prices[lp]>prices[rp])
                lp++;
            else
                rp++;
        }
        return res;
    }
};

File: 121. Best time to buy and sell stock/soln.py
"""
Sliding window 
Two pointer
"""

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        lp,rp = 0,0
        while rp<len(prices):
            profit = max(profit, prices[rp]-prices[lp])
            if prices[lp]>prices[rp]:
                lp +=1
            else:
                rp +=1
        return profit